### PR TITLE
hop-on: send current sim. task instead of next inside of sim. step

### PR DESCRIPTION
**Pull request by VDanielEdwards**
## Subject

This change fixes an issue with hopping onto running time-synchronized simulations with a participant with an autonomous lifecycle.

The bug can cause a participant to release it's sim step 'early' and cause other participants to execute their next simulation step too early.

**TODO**:
 - [x] Rework the log messages

## Developer checklist (address before review)

- [ ] Changelog.md updated
- [ ] Prepared update for depending repositories
- [ ] Documentation updated (public API changes only)
- [ ] API docstrings updated (public API changes only)
- [ ] Rebase &rarr; commit history clean
- [ ] Squash and merge &rarr; proper PR title

### CODE REVIEW

File: SilKit/source/services/orchestration/TimeSyncService.cpp [hunk 1]
```diff
@@ -370,7 +370,17 @@ TimeSyncService::TimeSyncService(Core::IParticipantInternal* participant, ITimeP
                                   "Participant \'{}\' is joining an already running simulation. Resending our "
                                   "NextSimTask.",
                                   descriptorParticipantName);
-                            SendMsg(_timeConfiguration.NextSimStep());
+
+                            if (GetTimeSyncPolicy()->IsExecutingSimStep())
```
Comment (MariusBgm) [inline SilKit/source/services/orchestration/TimeSyncService.cpp:374]: I wonder how we could test this with an integration test. we would have to add some nested participant, that requests nextSimTask while another one is hanging in SimulationStepAsync. Would that be feasible?

**Review by MariusBgm — APPROVED** [review]
LGTM! however we should maybe add a test for this edge case

File: SilKit/source/services/orchestration/TimeSyncService.cpp (hunk 1 above)
Comment (VDanielEdwards) [reply SilKit/source/services/orchestration/TimeSyncService.cpp:374]: The problematic behavior occurs when a participant performs a hop-on while the simulation step of another participant is still executing. Problem is that we would need to see all the `NextSimTask` messages, since under 'normal circumstances' – i.e., not having nested participants with weird synchronization between them – the system won't stop.

Comment (VDanielEdwards) [reply SilKit/source/services/orchestration/TimeSyncService.cpp:374]: We might be able to put it into a unit-test though. I'll have a look.

Comment (VDanielEdwards) [reply SilKit/source/services/orchestration/TimeSyncService.cpp:374]: Added two unit tests that ensure that the current (when hop-on happends during a simulation step) or the next (when hop-on happends outside of a simulation step) timestamp is used for the 'informative' message to the participant that is 'hopping on'.

Comment (MariusBgm) [reply SilKit/source/services/orchestration/TimeSyncService.cpp:374]: Thanks! looks good
